File: inspect_db.py
import json
import os

db_path = './server/src/data/db.json'
output_path = './output.txt'

try:
    with open(db_path, 'r', encoding='utf-8') as f:
        db = json.load(f)
    
    products = db.get('products', [])
    log_content = f"Total products: {len(products)}\n"
    for i, p in enumerate(products[:10]):
        log_content += f"Product {i}:\n"
        log_content += f"  ID: {p.get('id')}\n"
        log_content += f"  Brand: {p.get('brand')}\n"
        log_content += f"  Model: {p.get('model')}\n"
        log_content += f"  Price: {p.get('sellingPrice')}\n"
        log_content += f"  Images: {p.get('images')}\n"
        log_content += f"  ImageUrls: {p.get('imageUrls')}\n"
    
    with open(output_path, 'w', encoding='utf-8') as f:
        f.write(log_content)
    print("Success python inspect")
except Exception as e:
    with open(output_path, 'w', encoding='utf-8') as f:
        f.write(f"Error: {str(e)}")
    print(f"Error: {e}")

File: inspect_db.js
const fs = require('fs');
try {
  const db = JSON.parse(fs.readFileSync('./server/src/data/db.json', 'utf8'));
  const targetId = 'gGuoDKNxWmSeeJe785j2u';
  const product = db.products.find(p => p.id === targetId || p._id === targetId);
  const logContent = 'Target Product:\n' + JSON.stringify(product, null, 2) + '\n\n' +
                     'All Products:\n' + JSON.stringify(db.products.map(p => ({ id: p.id, brand: p.brand, laptopName: p.laptopName, price: p.price, sellingPrice: p.sellingPrice, purchasePrice: p.purchasePrice, imageUrls: p.imageUrls })), null, 2);
  fs.writeFileSync('./output.txt', logContent, 'utf8');
  console.log('Inspection success!');
} catch (e) {
  fs.writeFileSync('./output.txt', 'Error: ' + e.message, 'utf8');
  console.error('Inspection error:', e.message);
}
